src/components/CalendarMonthGrid.jsx

```diff
@@ -73,6 +76,23 @@ export default class CalendarMonthGrid extends React.Component {
     this.container.addEventListener('transitionend', this.onTransitionEnd);
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { initialMonth, numberOfMonths } = nextProps;
+    const { months } = this.state;
+
+    const hasMonthChanged = !this.props.initialMonth.isSame(initialMonth, 'month');
+    const hasNumberOfMonthsChanged = this.props.numberOfMonths !== numberOfMonths;
+    if (hasMonthChanged || hasNumberOfMonthsChanged) {
+      if (initialMonth.isAfter(this.props.initialMonth)) {
+        months.shift();
```

Comment:
This is mutating the state array directly, rather than using setState - is this intentional? Can we avoid the mutation?